wasmtime-wasi/src/host_impls.rs

```diff
@@ -1,5 +1,63 @@
+use super::host;
 use super::wasm32;
 
+pub unsafe fn wasmtime_ssp_args_get(
+    argv_environ: *mut host::argv_environ_values,
+    argv: *mut *mut host::char,
+    argv_buf: *mut host::char,
+) -> wasm32::__wasi_errno_t {
```

Comment:
Do we actually need to keep the C signatures? Or, as you pointed out, could we replace with Rust struct and chuck the `unsafe` out? For example, at a bare minimum, I guess we could pass `argv` as `&[*mut host::char]` instead of `*mut *mut host::char`. @sunfishcode thoughts on this?

Reply:
I agree, it does make sense to move to more Rust-idiomatic interfaces here.